File: src/test/java/com/vielheit/core/UserRepositoryTest.java
package com.vielheit.core;

import com.vielheit.CoreApplication;
import com.vielheit.core.domain.User;
import com.vielheit.core.repository.UserRepository;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.TestPropertySource;
import org.springframework.test.context.junit4.SpringRunner;

import static org.junit.Assert.assertEquals;

/**
 * Created by admin on 5/20/17.
 */
@RunWith(SpringRunner.class)
@TestPropertySource("classpath:application-test.properties")
@ContextConfiguration(classes = CoreApplication.class)
public class UserRepositoryTest {

    @Autowired
    private UserRepository userRepository;

    @Test
    public void findUser() {
        User user = new User();
        user.setActive(true);
        user.setEmailAddress("test@test.com");
        user.setFirstName("test");
        user.setLastName("test");
        user.setPassword("password");
        userRepository.save(user);

        User foundUser = userRepository.findByEmailAddress(user.getEmailAddress()).orElseGet(User::new);
        assertEquals(foundUser.getEmailAddress(), user.getEmailAddress());
        assertEquals(foundUser.getId(), user.getId());
    }
}

File: src/test/kotlin/org/vielheit/core/repository/UserRepositoryTest.kt
package org.vielheit.core.repository

import org.junit.Assert
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired
import org.vielheit.core.BaseTest
import org.vielheit.core.domain.User

class UserRepositoryTest : BaseTest() {
    @Autowired
    lateinit var userRepository: UserRepository

    @Test
    fun findByEmail() {
        val user = User(active = true, firstName = "test", lastName = "user", email = "test@user.com", password = "")
        userRepository.save(user)
        val foundUser = userRepository.findByEmail("test@user.com")!!
        Assert.assertEquals("test@user.com", foundUser.email)
        Assert.assertEquals("test", foundUser.firstName)
        Assert.assertEquals("user",  foundUser.lastName)
    }
}